### PR TITLE
Card::Save/LoadSnapshot

### PR DESCRIPTION
**Pull request by audetto**
Add 2 virtual methods to `Card` to load and save snapshots.

### CODE REVIEW

**Comment by tomcw** [comment]
@audetto - I guess you saw the review comment I posted (above)... but just in case.

**Comment by audetto** [comment]
I have indeed not.
But, even looking everywhere I cannot find them.

**Comment by tomcw** [comment]
Ah, I had done "Start review", but hadn't done "Submit review" - my fault!

File: source/SaveState.cpp [hunk 1]
```diff
@@ -328,94 +328,76 @@ static void ParseSlots(YamlLoadHelper& yamlLoadHelper, UINT unitVersion)
...
+		if (slot == 0)
+		{
+			SetExpansionMemType(type);	// calls GetCardMgr().Insert() & InsertAux()
+			CreateLanguageCard();
+			bRes = GetLanguageCard()->LoadSnapshot(yamlLoadHelper, slot, cardVersion);
+		}
+		else
+		{
+			GetCardMgr().Insert(slot, type);
+			bRes = GetCardMgr().GetRef(slot).LoadSnapshot(yamlLoadHelper, slot, cardVersion);
```
[the remark this answers is not in the document]
Comment (audetto) [reply source/SaveState.cpp:398]: good point. I will remove it.

**Comment by audetto** [comment]
There are a few checks that cards can only be inserted in some slots.
Ignoring the settings GUI, are these checks really needed?

**Comment by tomcw** [comment]
>There are a few checks that cards can only be inserted in some slots.
Ignoring the settings GUI, are these checks really needed?

Yes (maybe): for some cards (eg. mouse & serial) you can only have 1 instance inserted. But sure, there are other ways to prevent multiple instances.

I think also Mockingboard (in _both_ slots 4 and 5) is a bit weird, because of this 2-slot requirement, so probably can't go anywhere else (easily).

Once we have a better GUI (#853) then we can remove these checks.

**Comment by tomcw** [comment]
Thanks @audetto - merged it to master.

**Pull request merged**